### PR TITLE
Actions section of Context tab in task editor too narrow #170

### PR DESCRIPTION
**Pull request by BeckerFrank**
fix for UI problem (issue #170)

### CODE REVIEW

File: mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java [hunk 1]
```diff
@@ -231,20 +231,27 @@ public void dispose() {
 		}
 	}
 
+	private int setNewMaxWith(int oldValue, Control imageControl, Control linkControl) {
```
Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:234]: It doesn't `set` anything, we can name it like `calculateMaxWidth`

Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:234]: Would you like to name arguments to be single word, like `existing`, 'image', `link`

File: mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java [hunk 2]
```diff
@@ -231,20 +231,27 @@ public void dispose() {
...
+		Point imageSize = imageControl.computeSize(SWT.DEFAULT, SWT.DEFAULT, true);
+		Point hyperlinkSize = linkControl.computeSize(SWT.DEFAULT, SWT.DEFAULT, true);
+		return imageSize.x + hyperlinkSize.x > oldValue ? imageSize.x + hyperlinkSize.x : oldValue;
```
Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:237]: Would you consider to extract `imageSize.x + hyperlinkSize.x` to a local variable (and inline `imageSize` and `hyperlinkSize`)?
In this case return statement will be much more readable:
```java
return required > existing ? required : existing
```

File: mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java [hunk 3]
```diff
@@ -231,20 +231,27 @@ public void dispose() {
...
+	}
+
 	private void createActionsSection(Composite composite) {
 		Section section = toolkit.createSection(composite,
 				ExpandableComposite.TITLE_BAR | ExpandableComposite.EXPANDED);
 		section.setText(Messages.ContextEditorFormPage_Actions);
 
+		int with = 0;
```
Comment (ruspl-afed) [inline mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:245]: this is an example of `temporal coupling`
in fact what we need is to ensure that after adding all the children to `sectionClient` its `widthHint` is big enough, right? So, all this code wants to be in the very end of the block, where you do
```java
sectionClientGridData.widthHint = with
```

**Review by ruspl-afed — APPROVED** [review]
LGTM. Only stylistic comments.

File: mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java (hunk 3 above)
Comment (merks) [reply mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:245]: There is more than one link and depending on translation, it's not clear which one is wider...

Adding this to the end of the method also seemed to do the trick:
```
		sectionGridData.widthHint = section.computeSize(SWT.DEFAULT, SWT.DEFAULT).x;
```
I.e., like this on Windows (10):

image

Comment (BeckerFrank) [reply mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java:245]: OK, so I add some more local variables

Comment (merks) [reply mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java]: It really seems sufficient and much simpler to do only this:

image

I.e., let the section as a whole compute its preferred size...

Comment (BeckerFrank) [reply mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java]: Job with this solution is https://ci.eclipse.org/mylyn/view/GitHub%20Orga/job/GitHub-mylyn-Orga/job/org.eclipse.mylyn/view/change-requests/job/PR-183/3/

When finished we should use "Squash Merge" to bring this into main.
@merks: Thanks this indeed much less code!

**Pull request merged**